11_FFTBloom/main.cpp

```diff
@@ -0,0 +1,1192 @@
+			const auto srcImages = srcImageBundle.getContents();
+			const auto kerImages = kerImageBundle.getContents();
+			if (srcImages.empty() or kerImages.empty())
+				return logFail("Could not load image or kernel!");
+			auto srcImageCPU = IAsset::castDown<ICPUImage>(srcImages[0]);
+			auto kerImageCPU = IAsset::castDown<ICPUImage>(kerImages[0]);
+			const auto srcImageFormat = srcImageCPU->getCreationParameters().format;
+			const auto kerImageFormat = kerImageCPU->getCreationParameters().format;
+
+			// Create views for these images
+			ICPUImageView::SCreationParams viewParams[2] =
+			{
+				{
+					.flags = ICPUImageView::E_CREATE_FLAGS::ECF_NONE,
+					.image = std::move(srcImageCPU),
+					.viewType = IImageView<ICPUImage>::E_TYPE::ET_2D,
+					.format = srcImageFormat,
+					.subresourceRange = {
+						.aspectMask = IImage::E_ASPECT_FLAGS::EAF_COLOR_BIT,
+						.baseMipLevel = 0u,
+						.levelCount = ICPUImageView::remaining_mip_levels,
+						.baseArrayLayer = 0u,
+						.layerCount = ICPUImageView::remaining_array_layers
+					}
+				},
+				{
+					.flags = ICPUImageView::E_CREATE_FLAGS::ECF_NONE,
+					.image = std::move(kerImageCPU),
+					.viewType = IImageView<ICPUImage>::E_TYPE::ET_2D,
+					.format = kerImageFormat,
+					.subresourceRange = {
+						.aspectMask = IImage::E_ASPECT_FLAGS::EAF_COLOR_BIT,
+						.baseMipLevel = 0u,
+						.levelCount = ICPUImageView::remaining_mip_levels,
+						.baseArrayLayer = 0u,
+						.layerCount = ICPUImageView::remaining_array_layers
+					}
+			}
+			};
+			const auto srcImageViewCPU = ICPUImageView::create(std::move(viewParams[0]));
+			const auto kerImageViewCPU = ICPUImageView::create(std::move(viewParams[1]));
+
+			// Using asset converter
+			smart_refctd_ptr<nbl::video::CAssetConverter> converter = nbl::video::CAssetConverter::create({ .device = m_device.get(),.optimizer = {} });
+			// We don't want to generate mip-maps for these images (YET), to ensure that we must override the default callbacks.
+			struct SInputs final : CAssetConverter::SInputs
+			{
+				inline uint8_t getMipLevelCount(const size_t groupCopyID, const ICPUImage* image, const CAssetConverter::patch_t<asset::ICPUImage>& patch) const override
+				{
+					return image->getCreationParameters().mipLevels;
+				}
+				inline uint16_t needToRecomputeMips(const size_t groupCopyID, const ICPUImage* image, const CAssetConverter::patch_t<asset::ICPUImage>& patch) const override
+				{
+					return 0b0u;
+				}
+			} inputs = {};
+			inputs.logger = m_logger.get();
+			nbl::asset::ICPUImageView* CPUImageViews[2] = { srcImageViewCPU.get(), kerImageViewCPU.get() };
+			
+
+			// Need to provide patches to make sure we specify SAMPLED to get READ_ONLY_OPTIMAL layout after upload
+			CAssetConverter::patch_t<ICPUImageView> patches[2] =
+			{
+				{
+					CPUImageViews[0],
+					IImage::E_USAGE_FLAGS::EUF_SAMPLED_BIT
+				},
+				{
+					CPUImageViews[1],
+					IImage::E_USAGE_FLAGS::EUF_SAMPLED_BIT
+				}
+			};
```

Comment:
if you make them written into `ICPUDescriptorSet` as SAMPLED_IMAGE they'll get the usage flags automagically 
https://github.com/Devsh-Graphics-Programming/Nabla-Examples-and-Tests/pull/144/files#r1836574984

Reply:
It works in the sense that they do get transitioned to `READ_ONLY_OPTIMAL`, but there's an issue: the `updateDescriptorSets` done by `reserve` is throwing that the write happens when the images are still in `LAYOUT::UNDEFINED`. Posterior writes I do don't complain so I believe the layout transition done by the converter is happening before the layout transition. Will investigate

Reply:
Manually providing patches doesn't solve this issue either

Reply:
Ok it's a bug, opened [an issue](https://github.com/Devsh-Graphics-Programming/Nabla/issues/798)